test/bidirectional_graph_test.rb

```diff
@@ -153,8 +153,14 @@ def test_neighbors
 
   def test_each_neighbor
     @edges.flatten.to_set.each do |v|
-      assert_equal @out_neighbors[v], @dg.each_out_neighbor(v).inject(Set.new) { |s, v| s << v }
-      assert_equal @in_neighbors[v], @dg.each_in_neighbor(v).inject(Set.new) { |s, v| s << v }
+      out_neighbors = Set.new
+      @dg.each_out_neighbor(v) { |n| out_neighbors << n }
+      assert_equal @out_neighbors[v], out_neighbors
+      assert_equal @out_neighbors[v], @dg.each_out_neighbor(v).inject(Set.new) { |s, vv| s << vv }
```

Comment:
```suggestion
```
I think this is a redundant assertion to the previous one. Or do I miss something?

Reply:
Maybe so. It passed with the bug and still passes with the bug fixed, so I just left it there.

Reply:
Hm, I do not understand, why the `each_in_neighbor` test succeeded with the bug. I understand why `each_out_neighbor` worked, because it is an alias.
The cause might be the inner working of `inject`, which I do not want to investigate more. I would therefore delete the inject assertions, because the test is clear enough without it.

By the way: We should also change base.rb#each_adjacent and each_vertex to:

```ruby

   def each_vertex(&block) # :yields: v
      raise NotImplementedError
    end

    def each_adjacent(v, &block) # :yields: v
      raise NotImplementedError
    end
```

The parameter block is missing currently.

Reply:
It may not be strictly necessary in all cases but I believe being explicit is better.

Reply:
I removed the tests with `inject()`.